docs/src/main/asciidoc/datasource.adoc

```diff
@@ -9,55 +9,120 @@ include::./attributes.adoc[]
+
+== TL;DR
 
-To complete this guide, you will need:
+This is a quick introduction to datasource configuration.
+If you want a better understanding of how all this works, this guide has a lot more information in the subsequent paragraphs.
 
-* less than 10 minutes
-* an IDE
-* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
-* Apache Maven 3.5.3+
+=== JDBC datasource
 
-== Creating the Maven project
+Add the `agroal` extension plus one of `jdbc-derby`, `jdbc-h2`, `jdbc-mariadb`, `jdbc-mssql`, `jdbc-mysql` or `jdbc-postgresql`.
 
-First, we need a new project. Create a new project with the following command:
+Then configure your datasource:
 
-[source,shell,subs=attributes+]
+[source, properties]
 ----
-mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
-    -DprojectGroupId=org.acme \
-    -DprojectArtifactId=agroal-quickstart \
-    -DclassName="org.acme.datasource.GreetingResource" \
-    -Dpath="/hello"
-cd agroal-quickstart
+quarkus.datasource.db-kind=postgresql
+quarkus.datasource.username=<your username>
+quarkus.datasource.password=<your password>
+
+quarkus.datasource.jdbc.url=jdbc:postgresql://localhost:5432/hibernate_orm_test
+quarkus.datasource.jdbc.min-size=4
+quarkus.datasource.jdbc.max-size=16
 ----
 
-This will generate:
+=== Reactive datasource
+
+Add either the `reactive-pg-client` or the `reactive-mysql-client` extension.
+
+Then configure your reactive datasource:
+
+[source, properties]
+----
+quarkus.datasource.db-kind=postgresql
+quarkus.datasource.username=<your username>
+quarkus.datasource.password=<your password>
+
+quarkus.datasource.reactive.url=postgresql:///your_database
+quarkus.datasource.reactive.max-size=20
+----
 
-* the Maven structure
-* a landing page accessible on `http://localhost:8080`
-* an example `Dockerfile` files for both `native` and `jvm` modes
-* the application configuration file
-* an `org.acme.datasource.GreetingResource` resource
-* an example integration test
+== Default datasource
 
-== Adding maven dependencies
+A datasource can be either a JDBC datasource, a reactive one or both.
+It all depends on how you configure it and which extensions you added to your project.
 
-Next, you will need to add the `quarkus-agroal` dependency to your project.
+To define a datasource, start with the following:
+
+[source, properties]
+----
+quarkus.datasource.db-kind=h2
+----
+
+The database kind defines which type of database you will connect to.
+
+We currently include these built-in database kinds:
+
+* Derby: `derby`
+* H2: `h2`
+* MariaDB: `mariadb`
+* Microsoft SQL Server: `mssql`
+* MySQL: `mysql`
+* PostgreSQL: `postgresql`, `pgsql` or `pg`
+
+Note that these database kinds only facilitate things.
+For instance, when we have a JDBC driver extension for a given kind,
+we resolve the JDBC driver to use automatically so you don't need to configure it yourself.
+If you want to use a database kind outside of this list, just use `other` or whatever name you prefer.
```

Comment:
It would be nice to add a `[source, properties]` for this case. For example, how do I connect to a DB2 DB?

Reply:
It's explained a bit later how you can define a specific driver.

I think it's clear enough. We can see if people complain.

Note that I don't want that to be too prominent as things won't work in native.

Reply:
Right, Explaining that a different JDBC driver from the list is possible but won't work in native is one of the things it would be nice to have in this document

Reply:
Well, feel free to contribute once this PR is in. I personally have a ton of other more pressing things to do today.

Reply:
Sure, do you mind if I push a commit adding this to your PR or should I do this in a separate PR?

Reply:
Better do it in a separate PR after this one is in. I really want that one in merged ASAP.